garak/buffs/paraphrase.py

```diff
@@ -7,8 +7,8 @@
 
 import garak.attempt
 from garak import _config
-from garak.generators.huggingface import HFCompatible
 from garak.buffs.base import Buff
+from garak.resources.api.huggingface import HFCompatible
 
 
 class PegasusT5(Buff):
```

Comment:
Just an observation not required for this PR, it looks like this class could benefit from a refactor to use `HFCompatible` and expose the `para_model_name` and `hf_args` as `DEFAULT_PARAMS`.

```suggestion
class PegasusT5(Buff, HFCompatible):
```

Reply:
This change is incomplete the class needs to be extended to consume the `HFCompatible` mixin in `_load_model()`:

`self.torch_device` moves to the standardized `self.device` and should be detected/populated from `hf_args["device"]` with a call to `self._select_hf_device()` in `_load_model()`:
``` Python
    def __init__(self, config_root=_config) -> None:
        self.max_length = 60
        self.temperature = 1.5
        self.num_return_sequences = 6
        self.num_beams = self.num_return_sequences
        self.tokenizer = None
        self.para_model = None
        super().__init__(config_root=config_root)
```

``` Python
   def _load_model(self):
        from transformers import PegasusForConditionalGeneration, PegasusTokenizer

        self.device = self._select_hf_device()
        model_kwargs = self._gather_hf_params(
            hf_constructor=PegasusForConditionalGeneration.from_pretrained
        )  # will defer to device_map if device map was `auto` may not match self.device

        self.para_model = PegasusForConditionalGeneration.from_pretrained(
            self.para_model_name, **model_kwargs
        ).to(self.device)
        self.tokenizer = PegasusTokenizer.from_pretrained(self.para_model_name)
```

Not an issue for this PR, however I suspect a few more items should likely be promoted to `DEFAULT_PARAMS`.  The `max_length`, `temperature`, `num_return_sequences`, and possibly `num_beams` if the value does not always have to be equal to `num_return_sequences` should likely be exposed a configurable.  Since `Fast` looks like it may also have similar items to promote I think that can be deferred.

Reply:
Agree. Thanks for the details. Will mark as ready for review when out of draft.

Reply:
Adding the `model_kwargs` part led to the paraphraser returning all blanks. Did the rest of the integration and added a test to catch this unwanted behaviour.

Reply:
Lets drop the `_gather_hf_params()` for now as the `PegasusForConditionalGeneration.from_pretrained()` looks like it is not handling the extra args in the way the current code is expecting, I suspect `device` vs `device_map` is also impacting the expectations.